### ISSUE TITLE
Add a code action to initialize or set structure fields in C

### ISSUE THREAD

**Issue by yyangdid** [clangd/clangd#1271]
First of all, thanks for this good extension. 👍 

I'm using Vscode + vscode-clangd + EIDE to write the C for STM32, It works very well.

I want a feature, but I don't know where to look for the answer, so try my luck here. :)

The feature:

There is a struct type:
``` C
typedef struct {
  uint8_t as;
  uint8_t tb;
  uint8_t kk;
  uint8_t la;
  // .... more members
} Gtid_t;
```
There is a variable:
``` C
Gtid_t Gg;
```
Is there a way to quickly generate the following code, to "expand" struct members to fill:
``` C
Gg.as = as;
Gg.tb = tb;
Gg.kk = kk;
Gg.la = la;
// .... more
```

I found some information [here](https://vi.stackexchange.com/questions/22346/is-it-possible-to-create-a-snippet-to-expand-struct-members-to-fill) for vim.

Any help for vscode, please tell me, thanks! :)

**Comment by HighCommander4** [comment]
Moved to clangd repo since support for a code action like this would have to be implemented on the server side.

For C++, a somewhat similar code action that exists is to generate a memberwise-constructor (you can then use code completion to get clangd to help you write a call to that constructor, with placeholders for the arguments).

For C, as an alternative to assignment, a code action could generate designated initializers, something like this:

```
Gtid_t Gg {
  .as = <placeholder>,
  .tb = <placeholder>,
  .kk = <placeholder>
};
```

**Comment by kanashimia** [comment]
Hello, I just had to fill a lot of structures from one to another, for H.264, and it was very annoying that clangd doesn't help with this. 

rust-analyzer has such code action to fill struct fields, which generates:
```
Foo {
   [perform code action: fill struct fields]
}
```
```
Foo {
   a: todo!(),
   b: todo!(),
}
```
It would be great of clangd supported similar for the designated initialisers in both C and C++

My suggestion is to do the following:
```
Foo foo = {
   [perform code action: fill struct fields]
}
```
```
Foo foo = {
   .a = {},
   .b = {},
}
```

Using `{}` as a placeholder should work in both C and C++ for pretty much all types? It is also easy to regex replace in the editor.
Although for C it was added only recently in C23, but should work okay if compiler is new enough, it will just produce a warning on older standards.
